Modified2

horobito opened the pull request:
* 타임라인 부터 pr 작성 시작

horobito committed:
* 테스트용 코드 임시 pr

horobito committed:
* 1) Feed 에 속해있던 timeline 검색을 새로운 도메인 timeLine 으로 할당
* 2) timeline 의 service 부분 구현

horobito committed:
* 표현층에 Feed 정보 전달하기 위한 DTO 생성

horobito committed:
* 시간표현 LocalDataTime 에서 Instant 로 타입 변경

horobito committed:
* 도메인 영역 보여내주는 대신, DTO 로 보내주도록 수정
 + 이와 관련 내용들 수정

horobito committed:
* 로그인된 User 정보 가져오기 위한 서비스 형성
  -> 수업자료 사용해서 적용

horobito committed:
* TimeLineService 에서 SpringSecurity 활용하여 User 가져오도록 수정

horobito committed:
* FriendShipService 에서 SpringSecurity 활용하여 User 가져오도록 수정

horobito committed:
* 아직 테스트 안하는 부분 임시 주석 처리

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 좋아요 누른사람 처리를 새로운 도메인 영역을 만들어 할당하게 함

horobito committed:
* 새로운 도메인 영역의 레이어 생성 시작

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 필요없는 코드 제거 및 User 정보 찾는 코드 변경

horobito committed:
* 필요없는 코드 제거

horobito committed:
* repository 생성

horobito committed:
* 기존 feed 에서 담당하고 있는 부분 가져옴

horobito committed:
* like 기능 구현중 (1) - 분기 작성

aerain commented on src/main/java/com/example/demo/preperredPerson/controller/PreferredPersonController.java:
```diff
@@ -0,0 +1,22 @@
+package com.example.demo.preperredPerson.controller;
```
패키지는 전부 소문자 작성이 원칙입니다.

horobito committed:
* 패키지 이름 변경

horobito committed:
* 열거 상수 추가 -> 싫어요 기능 추가 대비

horobito committed:
* PreferredPerson 생성 기능 추가

horobito committed:
* 구현에 필요한 기능 추가

horobito committed:
* service 층에서 응답위해 필요한 열거타입 생성

horobito committed:
* feed 와 댓글 중 어디의 preference 인지 설정하는 코드

horobito committed:
* 오타 수정

horobito committed:
* 필요한 열거상수 추가

horobito committed:
* Feed 좋아요 기능 전체적인 구현 완료

horobito committed:
* 선호 상태 반환기능 추가

horobito committed:
* 선호 상태 변경기능 추가

horobito committed:
* 좋아요 수 변경기능 추가  -> 새 객체 반환하도록 설정

horobito committed:
* PreferredPersonController Feed 부분 구현

horobito committed:
* PreferredPersonController Feed 부분 구현

horobito committed:
* PreferredPersonService 댓글 부분 구현 시작 & 중첩 코드 제거

horobito committed:
* 단어 오류 변경 [Logined -> logged]

horobito committed:
* 해당 feed 에서 comment 찾는 책임을
  CommentService 에 부여

horobito committed:
* UserSessionService 이용해서 유저 찾도록 변경

horobito committed:
* 필요한 열거상수 추가[COMMENT_NOT_FOUND]

horobito committed:
* 댓글 좋아요 기능 구현 중
 1) 피드 없을 경우
 2) 글 작성자가 친구가 아닐 경우
 구현 ,

horobito committed:
* 댓글 좋아요 기능 구현 중 [댓글 없을 경우] 구현

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -21,11 +20,12 @@
     @Transactional
-    public List<FriendDto> getMyFriends(int page, int size) {
-        User user = null;
+    public List<FriendDto> getMyFriends(int page, int size) throws AccessDeniedException {
+        User user = userSessionService.getLoggeddUser();
```
User가 강하게 결합되어있네요.
요거는 요렇게 더 일반화 시킬 수 있지 않을까요?

```suggestion
    public List<FriendDto> getFriendsByUserId(Long userId, int page, int size) {
    
```

aerain commented on src/main/java/com/example/demo/friend/service/FriendShipService.java:
```diff
@@ -21,11 +20,12 @@
-        User user = null;
+    public List<FriendDto> getMyFriends(int page, int size) throws AccessDeniedException {
+        User user = userSessionService.getLoggeddUser();
         Identfication userId = Identfication.create(user.getId());
         UserInfo userInfo = createUserInfo(user, userId);
```
이거 유저 도메인 녀석들이죠? 뭔가 여깄으면 안될거같은데

horobito replied:
저것들은 친구 도메인에 있는 것들,
 지금 저렇게 유저 도메인이랑 햇갈리는 것들 어떻게 대체해서 리팩토링 해야할지 고민중

aerain commented on src/main/java/com/example/demo/preferredperson/domain/PreferredPerson.java:
```diff
@@ -0,0 +1,56 @@
+@Setter(AccessLevel.PACKAGE)
+@Entity
+@NoArgsConstructor
+public class PreferredPerson {
```
선호하는 유저? 의 정의가 잘 와닿지 않는데, 코멘트로 설명 부탁합니다.

horobito replied:
모호한 이름 수정[preferred Person -> Preference Info]

horobito committed:
* comment 내 독립성 해치는 코드 제거 및
  도메인 이름 의미 명확하게 변경
  [PreferredPerson-> PreferenceInfo]

horobito committed:
*  도메인 이름 의미 명확하게 변경
  [PreferredPerson-> PreferenceInfo]

horobito committed:
*  도메인 이름 의미 명확하게 변경
  [PreferenceDocument ->PreferenceLocation]

horobito committed:
*  [좋아요 숫자]의 의미가 명확하게 전달되도록 변경
  [Preference->PreferenceCount]

horobito committed:
*  Feed 와 Comment 의 선호 정보 혼동될 수 있으므로, repository 메소드에 매개변수 추가

horobito committed:
*  댓글 관련 좋아요 정보 구현

horobito committed:
* 관계를 좀 더 명확히 나타내는 이름으로 변경
[UserInfo-> Friender, Friend->Friendee]
 및 User 도메인과 헷갈릴 수 있는 이름 변경
 [Identification -> PersonId, Name -> PersonName]

horobito committed:
* 수정된 이름 반영

horobito committed:
* 수정된 이름 반영

horobito committed:
* 수정된 이름 반영

horobito committed:
* 수정된 이름 반영

horobito committed:
* 오타 수정

horobito committed:
* 좀 더 간단하게 다대다 관계처럼 동작하도록 변경

horobito committed:
* 단순 띄어쓰기

horobito committed:
* 잠시 테스트 코드 짜보는 중
 -> FriendShipService 코드 변경이 필요한 것을 느낌

horobito committed:
*  FriendShipService 코드 User 에 좀 더 덜 의존하도록 변경중

horobito committed:
*  FriendShipService 코드 User 에 좀 더 덜 의존하도록 변경시도중

horobito committed:
* User 에 덜 의존하도록
  User 내부 정보 뽑아서 저장하는 방법으로 구현 시도

horobito committed:
* UserService 에 유저 찾는 메소드 구현 후 FriendShipService 에 적용

horobito committed:
* UserService 에 유저 찾는 메소드 구현 후 FriendShipService 에 적용 2

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 아무 관계도 아닌 유저끼리의 친구 생성 테스트 성공

horobito committed:
* 테스트 일부 코드 위치 변경

horobito committed:
* CRUD 각각 테스트 코드 작성해보기

horobito committed:
* Create 중 이미 요청 또는 수락된 경우 테스트

horobito committed:
* Create 중 수락하는 경우에 대한 테스트 코드

horobito committed:
* 오타 변경

horobito committed:
*userService 로 유저 찾도록 역할 위임

horobito committed:
*친구관계 삭제 테스트

horobito committed:
*잘못된 부분 수정

horobito committed:
*삭제 부분 테스트 구현

horobito committed:
*삭제 부분 테스트 구현 2.

horobito committed:
*친구 관계 잘못 구현된 부분 수정 위한
 열거형 생성

horobito committed:
* 친구관계 표현을 위한 열거형 생성 후,
  이에 맞게 코드 수정

horobito committed:
* 친구관계 표현을 위한 열거형 생성 후,
  이에 맞게 코드 수정

horobito committed:
* 테스트 코드 일단 패스 .. 나중에 다시 시도할 것

horobito committed:
* 삭제부분 반환 값 설정 완료

horobito committed:
* Controller 층에서 삭제요청에 대한 결과 값에 따른 응답 코드 설정 완료

horobito committed:
* Feed 부분 반환 값 설정 완료 - 좀 불안함

horobito committed:
* FriendShip 친구 리스트 가져오는 것
  반환 값 설정

horobito committed:
* 미사용 테스트 코드 내 오류 발생부분 제거

horobito committed:
* 친구 삭제기능  응답 코드 작성

horobito committed:
* [내게 온 친구요청 찾는 기능] 의
   응답 코드 작성

horobito committed:
* 필요없는 코드 제거

horobito committed:
* [내 타임라인 찾기] 응답 코드 작성

horobito committed:
* user 정보 userservice에서 찾도록 변경

aerain commented on src/main/java/com/example/demo/feed/service/FeedService.java:
```diff
@@ -2,103 +2,117 @@
-                .toArray(WriterId[]::new);
 
-        int length = friendIds.length;
+    public Object findFeedDetailByFeedId(Long id) throws AccessDeniedException {
```
Object 무엇..?

aerain commented on src/main/java/com/example/demo/feed/service/FeedService.java:
```diff
@@ -2,103 +2,117 @@
-        for (int i=0; i<length; i++){
-            feedsList.addAll(feedRepository.findFeedsByWriter_Id(friendIds[i]));
+        }else if (friendship.getFriendState()!= FriendShipState.ACCEPT){
+            return RequestResult.UNAUTHORIZED;
```
흠 이때는 그냥 예외를 던지는게 맞겠네요.

aerain commented on src/main/java/com/example/demo/feed/service/FeedService.java:
```diff
@@ -2,103 +2,117 @@
-        return feed;
-    }
 
+    public Object makeFeedByContents(String insertedContent) throws AccessDeniedException {
 
+        if (insertedContent.length()==0){
+            return RequestResult.BAD_REQUEST;
```
여기도 마찬가지입니다. BAD_REQUEST라는걸 던져서 괜히 반환형이 Object가 되면 불편해집니다.
차라리 예외를 던지게하세요

aerain reviewed (changes requested):
나머지도 다 마찬가집니다

horobito committed:
* [Comment 만들기] 응답 코드 작성

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 클래스 위치 변경 및 새로운 레이어 생성

horobito committed:
* 스프링 시큐리티에서 필요한 userDetails 를 받아오기 위한 service 기능 구현

horobito committed:
* UserDetailsService 구현 및 그에 필요한 코드들 작성중
(1)

horobito committed:
* UserDetailsService 구현 및 그에 필요한 코드들 작성중
(2)

horobito committed:
* WebSecurityConfig 구현중

horobito committed:
* 사용할 filter 틀 만듬

horobito committed:
* InputStream 의 한버 읽히면 닫히는 문제 해결 위한
  Wrapper 생성 위해 새로운 라이브러리 추가

horobito committed:
* InputStream 의 한버 읽히면 닫히는 문제 해결 위한
  Wrapper 생성

horobito committed:
* ObjectMapper 넣어줌

horobito committed:
* UserAuthenticationFilter 에 Object Mapper 넣어줌

horobito committed:
* 내부에 필드 하나 추가

horobito committed:
* Filter 에 토큰 가지고 인증하는 과정 추가
 - 사실 내가 보면서 짜고있는데, 이해 안됨. 그냥 아~주 대략적인 흐름만 이렇게 되겠거니 하는 상태로 적는중

horobito committed:
* 일단 짠다

horobito committed:
* 필요없는 도메인 제거

aerain commented on src/main/java/com/example/demo/security/infrastructure/UserAuthenticationFilter.java:
```diff
@@ -1,7 +1,14 @@
 package com.example.demo.security.infrastructure;
 
+import com.example.demo.user.controller.UserController;
```
인프라층에선 표현층을 의존하면 안됩니다.

horobito committed:
* 필요없는 코드 제거

horobito committed:
* 필터링 하지 않는다는 코드 추가

horobito committed:
* 응답코드 수정 - 그러나 아직 잘 모르겠음

horobito committed:
* 1. 필요없는 코드 제거 - BasicInfo.java, Feed Repository 내 메소드

2. @AttributeOverride 사용 - Friender, Friendee

3, @Transient 사용 -  Friender, Friendee, FriendShip 의
getter에 적용

4. @Enumrated 사용 - FriendShipState 의 열거타입 필드에 사용

5. controller 부분의 응답 코드 올바르게 적용
   - PreferenceInfoController.java, TimeLineController 에 적용

 6. PasswordEncoder 책임 다른 부분에 떠넘김

horobito committed:
* 필요없는 코드 제거

horobito committed:
* [피드 좋아요] 부분 반환 코드 변경

horobito committed:
* [피드 좋아요] 부분 반환 코드 다시 변경

horobito committed:
* [댓글 좋아요] 부분 반환 코드 변경

horobito committed:
* [친구목록 가져오기] 부분 반환 코드 변경

horobito committed:
* 로그인 기능 구현 시도

horobito committed:
* 필요없는 코드 제거 및 로그인 기능 구현중 - 갑작스럽게 forwardFriendShip 에서 오류가 난다???

horobito committed:
* 로그인 기능 구현시도 3

horobito committed:
* 테스트 코드 중 필요없는 부분들 제거

horobito committed:
* [친구 삭제 ] 중 1. 친구 관계가 없는 경우 테스트

horobito committed:
* [친구 삭제 ] 중 2. 친구 관계가 있는 경우 테스트 // 주의 = service 부분 반환 값 아직 변경 안함 - 테스트 코드 다 작성 후 할 것임

horobito committed:
* [친구 찾기] 테스트 1. 필수 요구사항 작성

horobito committed:
* 친구관계 찾기 테스트 생성 시도

horobito committed:
* 반환값 변경

horobito committed:
* [친구요청 확인 테스트] 1. 알맞은 요청 확인 테스트

horobito committed:
* [친구요청 확인 테스트] 1.1 페이지 번호 또는 사이즈 잘못된 경우 오류 발생하는지 확인하는 코드 추가

horobito committed:
* [친구요청 확인 테스트] 2. 조건에 따른 반환 값 여부 확인

horobito committed:
* 친구 삭제 시 필요한 기능 추가

horobito committed:
* [친구 목록 확인 테스트] - 조건에 맞게 불러오는지 확인 -> 성공

horobito committed:
* 교훈 : 테스트 실시할 메소드 안에는
matcher를 넣지 말자

horobito committed:
* [요청 수락] 테스트 성공

horobito committed:
* 내부 코드 약간 수정

horobito committed:
* 내부 코드 약간 수정

horobito committed:
* [게시글 좋아요 테스트 1]
 - 해당 feed 없을 경우 테스트

horobito committed:
* [게시글 좋아요 테스트 2] - 작성자와 친구 관계가 아닐 경우 테스트

horobito committed:
* [게시글 좋아요 테스트 3] - 내 게시글인 경우

horobito committed:
* 테스트에 필요한 몇몇 코드 수정

horobito committed:
* 중첩 코드 제거 위한 변수 추출

horobito committed:
* Feed 위한 Helper 적용

horobito committed:
* Feed 좋아요 테스트

horobito committed:
* [Comment 좋아요 테스트 1]
 - Feed 없을 경우

  + CommentHelper 생성

horobito committed:
* [Comment 좋아요 테스트 ] - 완료

horobito committed:
* [FeedService 테스트 시작]

horobito committed:
* [Feed] 에 못 넣은 필드 추가

horobito committed:
* 코드 수정

horobito committed:
* [Feed 만들기] 테스트를 위한 코드들 작성 및 수정

horobito committed:
* [Feed 만들기] 2. 올바른 요청 테스트 성공

horobito committed:
* [Feed 삭제하기] 1. <존재하지 않는 피드에 삭제 요청했을 경우> 테스트

horobito committed:
* [Feed 삭제하기] 2. <타인의 글에 삭제를 요청했을 경우> 테스트

horobito committed:
* [Feed 세부정보 확인하기] 1. 게시글 없을 때 2. 친구 관계 아닐 때 테스트
+ 기타 코드 반환형 수정

horobito committed:
* [Feed 세부정보 확인하기] 3. 게시글 상세 조회 테스트

+ 클래스들 이름 변경

horobito committed:
* [Feed 세부정보 확인하기] 3. <게시글 상세 조회 테스트 > 의 테스트 게시글에 comment 도 넣고 테스트 해봄

horobito committed:
* [comment 만들기]
1. comment 를 추가할 feed 가 없을 경우
2. feed 는 있지만, 내용이 없는 경우

horobito committed:
* [comment 만들기]
3. 작성자와 친구관계가 아예 없는 경우

horobito committed:
* [comment 만들기]
4. 작성자와 친구가 승인되지 않은 경우

horobito committed:
* [comment 만들기]
5. 댓글 생성 테스트

horobito committed:
* [Timeline 테스트] 1. 페이지 정보 또는 크기 잘못 입력

aerain commented on src/main/java/com/example/demo/feed/controller/CommentController.java:
```diff
@@ -14,14 +20,21 @@
+
+        try {
+            CommentDto result = commentService.makeCommentByFeedIdAndContents(feedId, contents);
+            ResponseEntity.ok().body(result);
```
이거 반환 안하는거같네요 🤔

aerain commented on src/main/java/com/example/demo/feed/domain/Writer.java:
```diff
@@ -26,6 +26,14 @@ public Writer(WriterId id, WriterName wrtName) {
         this.wrtName = wrtName;
     }
 
+    public Long getId(){
+        return this.id.getWriterId();
+    }
+
+    public String getName(){
+        return this.wrtName.getWriterName();
+    }
+
```
이건 없는 편이 좋을것 같네요

aerain commented on src/main/java/com/example/demo/feed/service/CommentService.java:
```diff
@@ -2,56 +2,86 @@
-        WriterName wrtName = WriterName.create(user.getUserBasicInfo().getUsername());
-        Writer writer = Writer.create(id, wrtName);
+        if (friendship==null || friendship.getFriendState()!= FriendShipState.ACCEPT){
+            throw new IllegalAccessException();
```
적절한 예외가 아닌것 같습니다?

aerain commented on src/main/java/com/example/demo/feed/service/FeedService.java:
```diff
@@ -2,103 +2,114 @@
+        if (myId.getId().equals(feed.getId())){
+            feed.delete();
+        }else {
+            throw new IllegalAccessException();
```
요것도 마찬가집니다.

aerain commented on src/main/java/com/example/demo/friend/controller/FriendShipController.java:
```diff
@@ -20,33 +21,55 @@
+        FriendShipResult result = friendShipService.deleteFriendShipRequest(friendId);
+        switch ((FriendShipResult)result){
+            case NEVER_REQUESTED:
+                ResponseEntity.status(HttpStatus.BAD_REQUEST);
```
반환이 필요없으면 작성하지 않아도 될 것 같습니다.

aerain commented on src/test/java/com/example/demo/friend/service/FriendShipCreateTest.java:
```diff
@@ -0,0 +1,179 @@
+                .thenReturn(friendee);
+
+        //then
+
+
```
..? 요건 검증이 없네요

aerain commented on src/test/java/com/example/demo/friend/service/FriendShipCreateTest.java:
```diff
@@ -0,0 +1,179 @@
+                .thenReturn(forwardFriendShip)
+                .thenReturn(backwardFriendShip);
+
+        //then
+
+        sut.create(1L);
+        System.out.println("forwardFriendShip의 상태 : " + forwardFriendShip.getFriendState());
+        System.out.println("backwardFriendShip의 상태 : " + backwardFriendShip.getFriendState());
```
요기도 검증이 없는것 같습니다.

aerain reviewed (approved):
테스트 코드 만족스럽네요
